src/model-viewer-base.ts

```diff
@@ -208,7 +223,12 @@ export default class ModelViewerElementBase extends UpdatingElement {
     // sure that the value has actually changed before changing the loaded flag.
     if (changedProperties.has('src') && this.src !== this[$scene].model.url) {
       this[$loaded] = false;
-      this[$updateSource]();
+      (async () => {
+        const updateSourceProgress = this[$progressTracker].beginActivity();
+        await this[$updateSource](
+            (progress: number) => updateSourceProgress(progress * 0.9));
```

Comment:
Why `* 0.9`? So it doesn't go to 1.0 until we're actually done (below)?

Reply:
Yeah